### PR TITLE
Deployment via KCP proposal

### PR DESCRIPTION
**Pull request by astefanutti**
Proposal documentation for #94.

### CODE REVIEW

File: docs/proposals/deployment_via_kcp.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# GLBC Deployment via KCP
+
```
Comment (maleck13) [inline docs/proposals/deployment_via_kcp.md:2]: Not sure how to add a comment to the diagram. I really like the diagram nice and clear. One thing I wonder is if it should show the syncer on the workload clusters?

File: docs/proposals/deployment_via_kcp.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+It is proposed to deploy the GLBC and its dependencies in a dedicated KCP workspace, and rely on the [transparent deployment feature][1] that KCP provides, so that GLBC controllers are synced and running in workload clusters, while transparently operating on the data plane that resides in KCP.
+
+This is illustrated in the following diagram:
+
+![Deployment via KCP](assets/deployment_via_kcp.svg)
+
+This entails:
+* The KCP syncer is deployed in the GLBC workload cluster with the default set of synced resources, i.e., `deployments.apps`, `secrets`, `configmaps`, `serviceaccounts`;
+* The _cert-manager_ deployment is tailored to run the minimal set of required controllers (using the `--controllers=issuers,certificaterequests-issuer-acme,certificaterequests-issuer-ca,certificates-readiness` option), synced into the workload cluster, and operates on the GLBC workspace, thanks to the transparent deployment feature;
+* The required _cert-manager_ APIs are deployed using KCP API management APIs (`APIResourceSchema`, `APIExport`), so that it'll be possible to leverage any _cert-manager_ service, that may eventually already be hosted in some KCP environments;
```
Comment (maleck13) [inline docs/proposals/deployment_via_kcp.md:37]: Is this done by us as part of deployment in the workspace? Do you have a definitive doc (even if in code) for when you should use of APIResourceSchema and APIExport

File: docs/proposals/deployment_via_kcp.md (hunk 1 above)
Comment (astefanutti) [reply docs/proposals/deployment_via_kcp.md:2]: Good idea, I've updated the diagram with the syncer added to the workload clusters.

File: docs/proposals/deployment_via_kcp.md (hunk 2 above)
Comment (astefanutti) [reply docs/proposals/deployment_via_kcp.md]: It's covered in the KCP P3 demo script: https://github.com/kcp-dev/kcp/blob/main/contrib/demo/prototype3-script/script-apibindings.sh. I've updated the actions to have it mentioned.

**Review by maleck13 — APPROVED** [review]

**Pull request merged**